ReactWindows/ReactNative.Shared/Views/View/ReactViewManager.cs

```diff
@@ -92,6 +99,192 @@ protected override BorderedCanvas CreateViewInstance(ThemedReactContext reactCon
             return new BorderedCanvas();
         }
 
+        #region drag and drop
+#if WINDOWS_UWP
+
+        /// <summary>
+        /// Enables the Canvas as a drop target.
+        /// </summary>
+        [ReactProp("allowDrop")]
+        public void SetAllowDrop(BorderedCanvas view, bool allowDrop)
+        {
+            view.AllowDrop = allowDrop;
+
+            if (allowDrop)
+            {
+                view.DragEnter += OnDragEnter;
+                view.DragOver += OnDragOver;
+                view.Drop += OnDrop;
+                view.DragLeave += OnDragLeave;
+
+            }
+            else
+            {
+                view.DragEnter -= OnDragEnter;
+                view.DragOver -= OnDragOver;
+                view.Drop -= OnDrop;
+                view.DragLeave -= OnDragLeave;
+            }
+        }
+
+        private async Task<JObject> GetDataTransferInfo(DataPackageView data, bool drop = false)
+        {
+            var files = new JArray();
+            var items = new JArray();
+            var types = new JArray();
+
+            if (data.Contains(StandardDataFormats.StorageItems))
+            {
+                foreach (var item in await data.GetStorageItemsAsync())
+                {
+                    try
+                    {
+                        var file = item as StorageFile;
+
+                        if (file != null)
+                        {
+                            var props = await file.GetBasicPropertiesAsync();
+                            var type = file.ContentType;
+                            var path = drop ? "file://cache/" + GetAccessToken(file) : "";
+
+                            files.Add(new JObject
+                            {
+                                { "name", file.Name },
+                                { "size", props.Size },
+                                { "type", type },
+                                { "uri", path }
+                            });
+
+                            items.Add(new JObject
+                            {
+                                { "kind", "file" },
+                                { "type", type }
+                            });
+
+                            types.Add(type);
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        System.Diagnostics.Debug.WriteLine("GetDataTransferInfo: " + ex);
+                    }
+                }
+            }
+
+            return new JObject
+            {
+                { "files", files },
+                { "items", items },
+                { "types", types }
+            };
+        }
+
+        private string GetAccessToken(StorageFile file)
```

Comment:
the size and complexity of these changes make them inappropriate for a #if, in my opinion. can we push down the UWP-specifics in to a child class  (UwpReactViewManager) and parameterize its type to still allow discovery of ReactMethod annotations correctly? This would help the netstandard effort as well, actually.

Reply:
I agree that this change is too complex for an ifdef. Are you suggesting to crate a project like ReactNative.UWP (similar to ReactNative.Net46)?

Reply:
I've added WPF and UWP specific view managers.